### ISSUE TITLE
Team Chat

### ISSUE THREAD

**Issue by dsoutha3** [dsoutha3/Asteroids#1]
To assist in collaboration.

**Comment by dsoutha3** [comment]
I have all the images needed and everything. I am just having push issues.

**Comment by vcordes1** [comment]
sounds cool we can work them out tomorrow

**Comment by dsoutha3** [comment]
Hey guys, IK we are all busy, and we have some time until the games deadline, but we should all get in gear. It has been a couple of weeks, and I don't see much activity. Lets pull this together, so we can come up with a semi-complete game.

**Comment by Dwoodco2** [comment]
Yeah, things have been pretty busy these past few weeks but you do make a good point. I was planning on staying behind tomorrow and working on the project. I think we should get as much of it done as we can before next week. Don't we have all of next week off?

**Comment by dsoutha3** [comment]
Yes we do, but I won't be there tomorrow. My girlfriend has court. I have got a lot done though, I have the first wave of asteroids deploying in random spots, and moving at random velocitys, plus wrapping around the screen. I just can't do collision detection without a player class that shoots. Do a pull, then compile, then check it out. Then check out the source code:)I will work on the Alien computer spaceship, while you guys work on the player tomorrow.

**Comment by dsoutha3** [comment]
I just fixed the bug where it didn't read every key press at the right time by adding an SDL_Delay between the blitzed space-ship and the keypress. Also, I gave it a velocity while they key is down.

**Comment by dsoutha3** [comment]
You can find the movement event right below the collision detection screen wrap code.

**Comment by Dwoodco2** [comment]
Alright will pull the updates I have been working on the start menu should have something pushed by tomorrow.

**Comment by dsoutha3** [comment]
sweet

**Comment by Dwoodco2** [comment]
So for  the start menu we talked about an idea on Thursday its going to something like this

> > START
> >       SHIP SELECT (Not putting into menu yet haven't 100% decide if we want yet)
> >       CHEATS (Again not in menu yet)
> >       OPTIONS (For minor things like sound if we have time)
> >       EXIT

The > will be an image of the original asteroids ship that goes up and down next to the words to act as a pointer for the current option. I drew the images in paint but I'm not sure how to get them into the images folder we have set up from home, at the lair I can just drag and drop them in.

**Comment by Dwoodco2** [comment]
Also if you have any color scheme ideas for the words and the actual start menu right now its just black and white with a grey background, the font is in this alien tron like text.

**Comment by dsoutha3** [comment]
If you don't have a linux/unix OS, Get on putty (available on homepage at lab46.com.  ssh Dwooco2@lab46.corning-cc.edu. Create a folder with the command "mkdir foldername" . Then go in the folder and use either "hg clone repopath" or "git clone repopath" you can find the path here on git. Then you may email the images to your alpine mail . Finally, use the commad "mv path to image folder in repository. Then just add, commit, and push.

**Comment by dsoutha3** [comment]
green on black sounds cool, and those would be good additions.

**Comment by dsoutha3** [comment]
To whomever it doing the rotation on the ship, I have found some good info for you on stack overflow. This is probably what should be done next asap, so I can work on the asteroids collision detection.........

With SDL you have a few choices.

```
Rotate all your sprites in advance (pre-render all possible rotations) and render them like you would any other sprite. This approach is fast, but uses more memory and more sprites. As @Nick Wiggle pointed out, RotSprite is a great tool for generating sprite transoformations.

Use something like SDL_gfx to do real-time rotation/zooming. (Not recommended, very slow)

Use SDL in OpenGL mode and render your sprites to primitives, applying rotation to the primitives.
```

Option 3 is probably your best bet because you gain all of the advantages of using OpenGL. It's really up to you how to want to do it. It's also a possibility that you can load your sprites, perform all rotation calculations with SDL_gfx and then save the rotated versions to a SDL_Surface in memory.

EDIT: In response to your comment I would recommend checking out this GPWiki tutorial about using OpenGL with SDL. It doesn't cover rotation but it does cover basic set up and drawing. There is an OpenGl function, glRotatef, which can be useful in your case. A quick search brought back this little tidbit which could also be helpful.

**Comment by Dwoodco2** [comment]
So I was able to get the images added to the image folder. I forgot I had filezilla installed on my computer which lets me access the lab46 files like we can in the lair. For the image itself paint doesn't let remove the background around the image itself. For now I just made the background completely black until we can go back to the LAIR where I can use  the image editor there to make the image around maybe 300 by 400.

For the rotating we talked a little about that in class last week but didn't make any 100% decisions.
So things we have left pretty much are.

> Ship movement(Rotating mostly)
>   Collision detection
>   The rest of the smaller meteors and splitting into smaller ones (Goes with Collision)
>   Then a couple of minor tweaks and extra features added in
> - Anything I am missing

**Comment by dsoutha3** [comment]
Yeah, I am working on getting the alien shooting tonight.

**Comment by dsoutha3** [comment]
You didn't push the images.

**Comment by Dwoodco2** [comment]
Yeah the images I had originally created didn't work out very well. I had to recreate them but they take up smaller space and actually work in the images folder now. 

For the actual menu I basically set it up as a while loop for now that sets the game to start when an int is equal to a certain number. I'm going to use this int to start different while loops for the different options that can take place in the menu.

Then well in those while loops I am going to set up a key to break the while loop and take you back to the main menu to start the game or exit.

**Comment by Dwoodco2** [comment]
I can't push anything to the repo its doing that stupid thing where it tells me to push before I pull then tells me to pull before I push... It says I have to remove the asteroids.cc file from my folder to pull anything

**Comment by dsoutha3** [comment]
Why not just do the menu by mouse click?

**Comment by dsoutha3** [comment]
This is because we are working on the same head at the same time, simply
move your copy, pull the newest version, copy and paste to append your
changes from your moved file, add, commit, and push.

**Comment by Dwoodco2** [comment]
Yeah, I was planning on just moving it. For the mouse click I didn't even really think of using that the menu I will have to read up more on using the mouse. I've also been looking into the collision detection it isn't that difficult.

Next week I will most likely be spending every single day in the LAIR working on this. Do you know how late he usually keeps the LAIR open till?

**Comment by Dwoodco2** [comment]
Also for the mouse didn't he want us to have 2 different movement options for the games? We could use the mouse as the other one and the arrow keys as one.

**Comment by dsoutha3** [comment]
I don't recall him saying that, even so, it may be kind of akward to move
around a ship with a mouse. Especially a ship that is not meant to move as
quick as you can move our mouse.

**Comment by Dwoodco2** [comment]
Hmm I thought someone said that maybe not, I don't think he will mind if the game works all around.

I was also able to get a linux computer set up and am able to handle SDL2 at home much much easier now.

**Comment by dsoutha3** [comment]
Good, and idk maybe, but yeah right now what I really need for somebody to
do is the ships rotation with SDLK_LEFT and right, It would be horribly
redundant for large abundance of hard work already put in, not have enough
time to make playable.

**Comment by Dwoodco2** [comment]
I think Vincent said he was working on the ship and its movement. I can start working on it though I looked into the rotation a while ago well we were in class and talked it over with him.

**Comment by dsoutha3** [comment]
It is cool if he does it, but he hasn't yet. and we only have about 15
days, which is just about how long it will take for tweaks, collision
detection, and shooting. At least one wave is my goal, and f it is not done
in a of couple days, I don't see even one wave even getting done.

**Comment by Dwoodco2** [comment]
Yeah, I will start working on the ship rotating instead, I moved all the menu changes over to a separate file for now since a menu isn't really needed right now. Even Matt said the original Asteroids game just started right up with no menu.

**Comment by dsoutha3** [comment]
Thanks so much. We will get a menu up after, don't worry. I can help on
that too after.

**Comment by dsoutha3** [comment]
Btw, try more of to rotate it with a function, or tools like I posted up
previously. I hear if you try to do it all with just math, it will look
horrible.

**Comment by Dwoodco2** [comment]
Got all the libraries and files needed to combile our game installed on my computer I got it compiled and was able to load what we have in the game so far. I will definetly have something working with the rotating soon since I can actually test it on my own now.

**Comment by Dwoodco2** [comment]
Well working on the rotation and everything I loaded the game again to test something and looking at it the player ship does seem kind of large for the overall size of the game, I think we should shrink the player ship down a little more isn't it suppose to be smaller than the large meteors and the alien ship.

I am off to work right now but I almost have the ship rotating and moving, I will test it again when I get back tonight from work then push it. I will start working on the ship firing with the "SPACE" key next. I just have to the current degree of the ship and have the bullet shoot in a straight line in that direction when you hit the space key.

**Comment by dsoutha3** [comment]
Yup the key is to make the bullet just follow the tip of the ship invisibly, or always hold the values of the tip of the ship and then make the bullet visible from where it is, or put it at the value you have been holding for it and give it a velocity that is dependent on the ships direction.

**Comment by dsoutha3** [comment]
I just did a lot with the images. Pull and check it out.

**Comment by dsoutha3** [comment]
Thank god, got rid of that meatball looking asteroid.

**Comment by Dwoodco2** [comment]
So I got Matt to help me pull the changes you made. I ended up having to remove the entire folder and clone from the website again but I still got something different from yours. He said it looks like you didn't push the changes with the alien ship shooting.

**Comment by dsoutha3** [comment]
I have, and it says everything pushed is up to date. Do you have the latest version of funcs in the include file included? The shooting lasers are dependent on the timer class I put in there. Go to inc/func and check. Otherwise, the bullets will never shoot.

**Comment by dsoutha3** [comment]
Do the images look normal now?

**Comment by Dwoodco2** [comment]
They are still blue/purple. The alien ship doesn't shoot either. I moved the project to a different folder and cloned the project from github earlier in class to see if It just wasn't pulling the changes. I am still getting something different I tried to pull here at home also and it says its up to date.

**Comment by dsoutha3** [comment]
That is very weird.

**Comment by dsoutha3** [comment]
Well, ig push your changes up, I will see what I have and pull.

**Comment by dsoutha3** [comment]
It needs to be the difference of systems or idk, I just made a new directory and cloned and it works perfectly.

**Comment by Dwoodco2** [comment]
I have no idea why but it is still saying everything is up to date when I git push... I moved all of my changes over to the main file and it still says everything is up to date...
